src/main/java/javaserver/RequestParser.java

```diff
@@ -5,21 +5,44 @@
 	public static Request createRequest(String rawRequest) {
 		String requestMethod = getRequestMethod(rawRequest);
 		String requestURI = getRequestURI(rawRequest);
-		return new Request(requestMethod, requestURI);
+		String requestData = getRequestData(rawRequest);
+		return new Request(requestMethod, requestURI, requestData);
 	}
 
 	public static String getRequestMethod(String rawRequest) {
-		return splitRequest(rawRequest)[0];
+		return rawRequest.split("\\s")[0];
 	}
 
 	public static String getRequestURI(String rawRequest) {
 		try {
-			return splitRequest(rawRequest)[1];
+			return rawRequest.split("\\s")[1];
 		} catch (ArrayIndexOutOfBoundsException e) {
 			return "";
 		}
 	}
 
+	public static String getRequestData(String rawRequest) {
```

Comment:
This is getting called every time you create a response.  Maybe we can check if there is data to be parsed first by having a `isData` boolean method.